searx/engines/recoll.py

```diff
@@ -0,0 +1,79 @@
+"""
+ Recoll (local search engine)
+
+ @using-api   yes
+ @results     JSON
+ @stable      yes
+ @parse       url, content, size, abstract, author, mtype, time, filename, label
+"""
+
+from json import loads
+from searx.url_utils import urlencode
+from datetime import date, timedelta
+
+# engine dependent config
+paging = True
+time_range_support = True
+
+# parameters from settings.yml
+base_url = None
+search_dir = ''
+dl_prefix = None
+
+
+def get_time_range(time_range):
+    sw = {
+        'day': 1,
+        'week': 7,
+        'month': 30,
+        'year': 365
+    }
+
+    offset = sw.get(time_range, 0)
+    if not offset:
+        return ''
+
+    return (date.today() - timedelta(days=offset)).isoformat()
+
+
+# do search-request
+def request(query, params):
+    search_after = get_time_range(params['time_range'])
+    search_url = base_url + 'json?query={query}&page={page}&after={after}&dir={dir}&highlight=0'
+
+    params['url'] = search_url.format(query=urlencode({'q': query}), page=params['pageno'],
```

Comment:
I would urlencode variable `dir`, too. If someone has weird directory names, it shouldn't break the query.

To me it seems like this URL is incorrect. If the URL encoded query is substituted into this format string, the following is returned: `https://my.recoll.me/json?query=q=mytestquery&page=1&after=&dir=&hightlight=`. The part `query=q=mytestquery` doesn't seem right.

Also, as a note, I would urlencode everything. It's up to you if you change it or not.

```python
search_url = base_url + 'json?{query}&highlight=0'
search_url.format(query=urlencode({
    'q': query,
    'page': params['pageno'],
    'after': search_after,
    'dir': search_dir,
}))
```